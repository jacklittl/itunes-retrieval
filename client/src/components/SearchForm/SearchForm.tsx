import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { fetchData } from 'redux/form/form.thunk';

import {
  makeStyles,
  Container,
  Grid,
  InputLabel,
  MenuItem,
  FormControl,
  Select,
  Button,
  TextField,
} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  formControl: {
    width: 100 + '%',
  },
}));

const mediaOptions = [
  {value: 'movie', label: 'Movie' },
  {value: 'podcast', label: 'Podcast' },
  {value: 'music', label: 'Music' },
  {value: 'musicVideo', label: 'Music Video' },
  {value: 'audioBook', label: 'Audio Book' },
  {value: 'shortFilm', label: 'Short Film' },
  {value: 'tvShow', label: 'TV Show' },
  {value: 'software', label: 'Software' },
  {value: 'all', label: 'All' }
];

type Query = {
  media?: string,
  term?: string
}

const SearchForm = () => {
  const dispatch = useDispatch();
  const [query, setQuery] = useState<Query>({
    media: '',
    term: ''
  });

  const classes = useStyles();

  const onChange = ( data, field ) => {
    if ( field === 'term' ) setQuery({...query, term: data });
    else setQuery({...query, media: data });
  }

  const onSubmit = () => dispatch( fetchData( query ) );

  console.log( query )

  return (

    <form 
      name="filters" 
      className="section" 
    >

      <Container maxWidth="lg">
            
          <Grid container justify="center" spacing={2}>
                          
            <Grid item xs={12} sm={3}>

              <FormControl variant="filled" className={classes.formControl}>

                <InputLabel id="demo-simple-select-filled-label">Title</InputLabel>

                <Select
                  value={query.media}
                  labelId="demo-simple-select-filled-label"
                  id="demo-simple-select-filled"
                  onChange={value => onChange(value.target.value, 'media')}
                >
                
                  {mediaOptions.map( mediaOption => (

                    <MenuItem key={mediaOption.value} value={mediaOption.value}>{mediaOption.label}</MenuItem>

                  ))}

                </Select>

              </FormControl>

            </Grid>

            <Grid item xs={12} sm={7}>

              <FormControl variant="filled" className={classes.formControl}>

                <TextField 
                  id="term"
                  name="term"
                  value={query.term}
                  label="Search Text"
                  onChange={e => onChange(e.target.value, 'term')}
                  variant="filled"
                />

              </FormControl>

            </Grid>

            <Grid item xs={5} sm={2}>

              <FormControl variant="filled" className={classes.formControl}>
                
                <Button variant="contained" onClick={() => onSubmit()} color="primary" size="large">Search</Button>
              
              </FormControl>

            </Grid>

        </Grid>

      </Container>

    </form>
  )
};

export default SearchForm;