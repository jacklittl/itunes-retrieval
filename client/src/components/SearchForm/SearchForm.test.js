import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

import SearchForm from './SearchForm';

const mockStore = configureStore([]);
const store = mockStore({
  query: {
    media: "",
    term: ""
  }
});

configure({adapter: new Adapter()});

describe('Search Form Render', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <Provider store={store}>
        <SearchForm />
      </Provider>
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});