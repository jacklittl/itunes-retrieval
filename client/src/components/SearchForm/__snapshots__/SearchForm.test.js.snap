// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form Render snapshot renders 1`] = `
<form
  className="section"
  name="filters"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-1"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
            data-shrink={false}
            id="demo-simple-select-filled-label"
          >
            Title
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-filled-label demo-simple-select-filled"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
              id="demo-simple-select-filled"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-1"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink={false}
              htmlFor="term"
              id="term-label"
            >
              Search Text
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input"
                disabled={false}
                id="term"
                name="term"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-sm-2"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-1"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Search
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
